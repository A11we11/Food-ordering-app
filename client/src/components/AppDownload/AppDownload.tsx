import "./AppDownload.css";
import google from "../../assets/store/googleplay.png";
import apple from "../../assets/store/appstore2.png";

const AppDownload = () => {
  return (
    <section
      className="app-download"
      id="app-download"
      aria-label="App Download"
    >
      <p>
        For Better Experience Download <br /> QuikBite
      </p>
      <div className="app-download-platforms">
        <img src={google} alt="google image" />
        <img src={apple} alt="apple image" />
      </div>
    </section>
  );
};

export default AppDownload;
