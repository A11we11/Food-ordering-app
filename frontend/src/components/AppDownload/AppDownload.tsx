import "./AppDownload.css";
import google from "../../assets/store/googleplay.png";
import apple from "../../assets/store/appstore2.png";

const AppDownload = () => {
  return (
    <div className="app-download" id="app-download">
      <p>
        For Better Experience Download <br /> QuikBite
      </p>
      <div className="app-download-platforms">
        <img src={google} alt="" />
        <img src={apple} alt="" />
      </div>
    </div>
  );
};

export default AppDownload;
