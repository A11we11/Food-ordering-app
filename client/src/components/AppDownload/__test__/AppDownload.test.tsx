// AppDownload.test.tsx
import { render, screen } from "@testing-library/react";
import { vi, describe, it, expect } from "vitest";
import AppDownload from "../AppDownload";

// Mock CSS import
vi.mock("./AppDownload.css", () => ({}));

// Mock image imports - FIXED PATHS
vi.mock("../../../assets/store/googleplay.png", () => ({
  default: "mocked-assets/store/googleplay.png",
}));

vi.mock("../../../assets/store/appstore2.png", () => ({
  default: "mocked-assets/store/appstore2.png",
}));

describe("AppDownload Component", () => {
  it("renders with correct structure and id", () => {
    render(<AppDownload />);

    const container = screen.getByRole("region");
    expect(container).toHaveClass("app-download");
    expect(container).toHaveAttribute("id", "app-download");
  });

  it("displays the main text content", () => {
    render(<AppDownload />);

    // Check for the main text with line break
    expect(
      screen.getByText(/For Better Experience Download/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/QuikBite/i)).toBeInTheDocument();

    /*  // Alternative: Check for combined text content
    const textElement = screen.getByText(/For Better Experience Download/);
    expect(textElement).toHaveTextContent(
      "For Better Experience Download QuikBite"
    ); */
    // Matches combined text, ignoring whitespace/line breaks
    expect(
      screen.getByText(/For Better Experience Download\s+QuikBite/i)
    ).toBeInTheDocument();
  });

  it("renders both platform images", () => {
    render(<AppDownload />);

    const images = screen.getAllByRole("img");
    expect(images).toHaveLength(2);

    // Check Google Play image
    expect(images[0]).toHaveAttribute(
      "src",
      "mocked-assets/store/googleplay.png"
    );
    expect(images[0]).toHaveAttribute("alt", "google image");

    // Check App Store image
    expect(images[1]).toHaveAttribute(
      "src",
      "mocked-assets/store/appstore2.png"
    );
    expect(images[1]).toHaveAttribute("alt", "apple image");
  });

  it("applies correct CSS classes", () => {
    const { container } = render(<AppDownload />);

    expect(container.querySelector(".app-download")).toBeInTheDocument();
    expect(
      container.querySelector(".app-download-platforms")
    ).toBeInTheDocument();
  });

  it("has proper structure with platforms container", () => {
    const { container } = render(<AppDownload />);

    const platformsContainer = container.querySelector(
      ".app-download-platforms"
    );
    expect(platformsContainer).toBeInTheDocument();

    // Check that both images are inside the platforms container
    const imagesInContainer = platformsContainer?.querySelectorAll("img");
    expect(imagesInContainer).toHaveLength(2);
  });

  it("renders text with line break structure", () => {
    render(<AppDownload />);

    // The component uses <br /> tag, so we test that the structure allows for line breaks
    const textElement = screen.getByText(/For Better Experience Download/);
    expect(textElement.innerHTML).toContain("<br>");
  });

  // Accessibility tests
  it("has accessible structure", () => {
    render(<AppDownload />);

    // Main container should be identifiable
    const container = screen.getByRole("region");
    expect(container).toBeInTheDocument();

    // Images should be present (even with empty alt text)
    const images = screen.getAllByRole("img");
    images.forEach((img) => {
      expect(img).toHaveAttribute("alt");
    });
  });

  // Snapshot test (optional)
  it("matches snapshot", () => {
    const { container } = render(<AppDownload />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
