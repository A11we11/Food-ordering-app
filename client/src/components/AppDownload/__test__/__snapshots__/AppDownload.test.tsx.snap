// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppDownload Component > matches snapshot 1`] = `
<section
  aria-label="App Download"
  class="app-download"
  id="app-download"
>
  <p>
    For Better Experience Download 
    <br />
     QuikBite
  </p>
  <div
    class="app-download-platforms"
  >
    <img
      alt="google image"
      src="mocked-assets/store/googleplay.png"
    />
    <img
      alt="apple image"
      src="mocked-assets/store/appstore2.png"
    />
  </div>
</section>
`;
